test/Dialect/Substrait/literal.mlir

```diff
@@ -1,12 +1,36 @@
 // RUN: substrait-opt -split-input-file %s \
 // RUN: | FileCheck %s
 
+// CHECK:      substrait.plan version 0 : 42 : 1 {
+// CHECK-NEXT:   relation
+// CHECK:         %[[V0:.*]] = named_table
+// CHECK-NEXT:    %[[V1:.*]] = project %[[V0]] : tuple<si1> -> tuple<si1, !substrait.fixed_char<5>> {
+// CHECK-NEXT:    ^[[BB0:.*]](%[[ARG0:.*]]: tuple<si1>):
+// CHECK-NEXT:      %[[V2:.*]] = literal #substrait.fixed_char<"hello"> : !substrait.fixed_char<5>
+// CHECK-NEXT:      yield %[[V2]] : !substrait.fixed_char<5>
+// CHECK-NEXT:    }
+// CHECK-NEXT:    yield %[[V1]] : tuple<si1, !substrait.fixed_char<5>>
+
+substrait.plan version 0 : 42 : 1 {
+  relation {
+    %0 = named_table @t1 as ["a"] : tuple<si1>
+    %1 = project %0 : tuple<si1> -> tuple<si1, !substrait.fixed_char<5>> {
+    ^bb0(%arg : tuple<si1>):
+      %fixed_char = literal #substrait.fixed_char<"hello"> : !substrait.fixed_char<5>
```

Comment:
There's some redundancy here: The attribute name `#substrait.fixed_char` and the length of the string already fully define the type `!substrait.fixed_char<5>`. Do we want to remove it? I think that this could be done with our `TypeInferableAttrInterface`. If we don't do it now, please leave a TODO.